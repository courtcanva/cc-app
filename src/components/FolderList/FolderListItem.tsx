import { Box, Flex, HStack, Tag, IconButton, Text } from "@chakra-ui/react";
import Image from "next/image";
import FolderDeleteModal from "./FolderDeleteModal";
import React from "react";
import { format, parseISO } from "date-fns";

const FolderListItem = (props: any) => {
  const { courtId, courtName, designName, createdAt, image, courtType } = props.design;
  const createDate = format(parseISO(createdAt), "dd/MM/yyyy");

  return (
    <Flex
      key={courtId}
      width="232px"
      height="160px"
      background="white"
      marginBottom="18px"
      alignItems="center"
      justifyContent="space-between"
      cursor="pointer"
      flexDirection="column"
      onClick={() => props.handleCourtSelecting(courtId)}
      data-testid={courtId}
      _hover={{ border: "2px solid button.hover" }}
      opacity={!props.activateDesign || props.activateDesign === courtId ? "1" : "0.4"}
    >
      <HStack spacing={8}>
        <Box
          style={{
            minWidth: "140px",
            marginTop: "5px",
          }}
        >
          <Text variant="textFont">{designName}</Text>
          <Text variant="dateFont">created at {createDate}</Text>
        </Box>
        <IconButton aria-label="delete button" variant="deleteIconBtn" size="xs">
          <FolderDeleteModal />
        </IconButton>
      </HStack>
      <Box width="80%" height="100%" position="relative">
        {image && image.startsWith("http") && (
          <Image src={image.toString()} alt="Court image" layout="fill" objectFit="contain"></Image>
        )}
      </Box>
      <HStack spacing={8} style={{ margin: "4px" }}>
        <Tag size="sm" variant="courtName">
          {courtName.replace(/ /g, "")}
        </Tag>
        <Tag size="sm" variant="courtType">
          BASKETBALL
          {/* todo: need to update interface when tennis court is added  */}
        </Tag>
      </HStack>
    </Flex>
  );
};

export default FolderListItem;
