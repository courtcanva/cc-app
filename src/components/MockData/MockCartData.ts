import { ICartItem } from "@/interfaces/cartItem";

export const mockCartDataRaw: any = [
  {
    _id: {
      $oid: "632a6741439020e45d11aa2a",
    },
    quotation: "0.00",
    quotationDetails: [
      {
        color: "#7088B1",
        quantity: 71,
      },
      {
        color: "#E18E11",
        quantity: 57,
      },
      {
        color: "#B6B6B6",
        quantity: 59,
      },
      {
        color: "#834085",
        quantity: 0,
      },
      {
        color: "#2C4E8A",
        quantity: 323,
      },
    ],
    isExpired: true,
    user_id: "117321211547215290439",
    design: {
      designName: "Small Court example",
      courtSize: {
        name: "Small Court",
        length: 9000,
        width: 5000,
        threePointLine: 900,
        threePointRadius: 6600,
        centreCircleRadius: 1800,
        restrictedAreaLength: 5790,
        restrictedAreaWidth: 4800,
        sideBorderWidth: 0,
        lengthOfCorner: 1575,
        lineBorderWidth: 200,
      },
      tileColor: [
        {
          location: "threePoint",
          color: "#7088B1",
        },
        {
          location: "courtArea",
          color: "#E18E11",
        },
        {
          location: "topKeyArea",
          color: "#B6B6B6",
        },
        {
          location: "border",
          color: "#834085",
        },
        {
          location: "keyArea",
          color: "#2C4E8A",
        },
        {
          location: "circleArea",
          color: "#B6B6B6",
        },
      ],
    },
    image:
      "https://jr-bookinglet-2.s3.ap-southeast-2.amazonaws.com/court_canva_test/m8i7liw-KKm9Dl2OYwgi_.png",
    createdAt: {
      $date: {
        $numberLong: "1663723329368",
      },
    },
    updatedAt: {
      $date: {
        $numberLong: "1663723329368",
      },
    },
    isDeleted: false,
    __v: 0,
  },
  {
    _id: {
      $oid: "632a6761439020e45d11aa2d",
    },
    quotation: "0.00",
    quotationDetails: [
      {
        color: "#7088B1",
        quantity: 1273,
      },
      {
        color: "#E18E11",
        quantity: 2572,
      },
      {
        color: "#B6B6B6",
        quantity: 231,
      },
      {
        color: "#834085",
        quantity: 1000,
      },
      {
        color: "#2C4E8A",
        quantity: 624,
      },
    ],
    isExpired: false,
    user_id: "117321211547215290430",
    design: {
      designName: "Pro Full Court example",
      courtSize: {
        name: "Pro Full Court",
        length: 28000,
        width: 15000,
        threePointLine: 900,
        threePointRadius: 6600,
        centreCircleRadius: 1800,
        restrictedAreaLength: 5790,
        restrictedAreaWidth: 4900,
        sideBorderWidth: 1000,
        lengthOfCorner: 1575,
        lineBorderWidth: 200,
      },
      tileColor: [
        {
          location: "threePoint",
          color: "#7088B1",
        },
        {
          location: "courtArea",
          color: "#E18E11",
        },
        {
          location: "topKeyArea",
          color: "#B6B6B6",
        },
        {
          location: "border",
          color: "#834085",
        },
        {
          location: "keyArea",
          color: "#2C4E8A",
        },
        {
          location: "circleArea",
          color: "#B6B6B6",
        },
      ],
    },
    image:
      "https://jr-bookinglet-2.s3.ap-southeast-2.amazonaws.com/court_canva_test/m8i7liw-KKm9Dl2OYwgi_.png",
    createdAt: {
      $date: {
        $numberLong: "1663723361318",
      },
    },
    updatedAt: {
      $date: {
        $numberLong: "1663723361318",
      },
    },
    isDeleted: false,
    __v: 0,
  },
  {
    _id: {
      $oid: "632a6761439020e45d11aa2e",
    },
    quotation: "0.00",
    quotationDetails: [
      {
        color: "#7088B1",
        quantity: 1273,
      },
      {
        color: "#E18E11",
        quantity: 2572,
      },
      {
        color: "#B6B6B6",
        quantity: 231,
      },
      {
        color: "#834085",
        quantity: 1000,
      },
      {
        color: "#2C4E8A",
        quantity: 624,
      },
    ],
    isExpired: false,
    user_id: "117321211547215290431",
    design: {
      designName: "Pro Full Court example",
      courtSize: {
        name: "Pro Full Court",
        length: 28000,
        width: 15000,
        threePointLine: 900,
        threePointRadius: 6600,
        centreCircleRadius: 1800,
        restrictedAreaLength: 5790,
        restrictedAreaWidth: 4900,
        sideBorderWidth: 1000,
        lengthOfCorner: 1575,
        lineBorderWidth: 200,
      },
      tileColor: [
        {
          location: "threePoint",
          color: "#7088B1",
        },
        {
          location: "courtArea",
          color: "#E18E11",
        },
        {
          location: "topKeyArea",
          color: "#B6B6B6",
        },
        {
          location: "border",
          color: "#834085",
        },
        {
          location: "keyArea",
          color: "#2C4E8A",
        },
        {
          location: "circleArea",
          color: "#B6B6B6",
        },
      ],
    },
    image:
      "https://jr-bookinglet-2.s3.ap-southeast-2.amazonaws.com/court_canva_test/m8i7liw-KKm9Dl2OYwgi_.png",
    createdAt: {
      $date: {
        $numberLong: "1663723361318",
      },
    },
    updatedAt: {
      $date: {
        $numberLong: "1663723361318",
      },
    },
    isDeleted: false,
    __v: 0,
  },
];

export const mockCartData: ICartItem[] = mockCartDataRaw.map((cartItem: any) => {
  return {
    id: cartItem._id,
    user_id: cartItem.user_id,
    design: cartItem.design,
    quotation: cartItem.quotation,
    quotationDetails: cartItem.quotationDetails,
    image: cartItem.image,
    isExpired: cartItem.isExpired,
  };
});
