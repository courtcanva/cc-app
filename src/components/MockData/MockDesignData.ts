import { IDesignDetail } from "@/interfaces/design";

export const mockDesignData: any = [
  {
    courtId: "folderListItems-0",
    courtName: "full court",
    courtAreaXLength: 28000,
    courtAreaYLength: 15000,
    threePointLineToCourtEdgeLength: 900,
    cornerThreePointLineLength: 1575,
    threePointLineRadius: 6600,
    keyAreaWidth: 5790,
    keyAreaHeight: 4800,
    circleRadius: 1800,
    strokeWidth: 200,
    borderLength: 2000,
    customizeCourtAreaXLength: 0,
    customizeCourtAreaYLength: 0,
    designName: "Court Canva 0",
    createdAt: "2022-11-02T04:27:12.179+0000",
    updatedAt: "2022-11-05T08:57:32.220+0000",
    image:
      "https://courtcanva-image-node.s3.ap-southeast-2.amazonaws.com/preview-image/Bd3poCt9hQFcPRjkPJcXl.png",
  },
  {
    courtId: "folderListItems-1",
    courtName: "full court",
    courtAreaXLength: 28000,
    courtAreaYLength: 15000,
    threePointLineToCourtEdgeLength: 900,
    cornerThreePointLineLength: 1575,
    threePointLineRadius: 6600,
    keyAreaWidth: 5790,
    keyAreaHeight: 4800,
    circleRadius: 1800,
    strokeWidth: 200,
    borderLength: 1000,
    customizeCourtAreaXLength: 0,
    customizeCourtAreaYLength: 0,
    designName: "Court Canva 1",
    createdAt: "2022-11-05T04:27:12.179+0000",
    updatedAt: "2022-11-05T08:57:32.220+0000",
    image:
      "https://courtcanva-image-node.s3.ap-southeast-2.amazonaws.com/preview-image/Bd3poCt9hQFcPRjkPJcXl.png",
  },
];
