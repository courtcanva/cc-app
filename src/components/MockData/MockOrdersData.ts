import { IMyOrder } from "@/interfaces/order";
export const mockOrdersRowData: any = [
  {
    _id: "6370d83de71fd524e1c437b8",
    user_id: "6368e90574ac194077a36814",
    status: "completed",
    items: [
      {
        design: {
          designName: "Court Canva 1",
          courtSize: {
            name: "Pro Full Court",
            length: 28000,
            width: 15000,
            threePointLine: 900,
            threePointRadius: 6600,
            centreCircleRadius: 1800,
            restrictedAreaLength: 5790,
            restrictedAreaWidth: 4800,
            sideBorderWidth: 1000,
            lengthOfCorner: 1575,
            lineBorderWidth: 200,
          },
          tileColor: [
            {
              location: "threePoint",
              color: "#7088B1",
            },
            {
              location: "courtArea",
              color: "#E18E11",
            },
            {
              location: "topKeyArea",
              color: "#B6B6B6",
            },
            {
              location: "border",
              color: "#834085",
            },
            {
              location: "keyArea",
              color: "#2C4E8A",
            },
            {
              location: "circleArea",
              color: "#B6B6B6",
            },
          ],
          courtType: "basketball",
        },
        quotation: "67445.40",
        quotationDetails: [
          {
            color: "#7088B1",
            quantity: 1273,
          },
          {
            color: "#E18E11",
            quantity: 2572,
          },
          {
            color: "#B6B6B6",
            quantity: 231,
          },
          {
            color: "#834085",
            quantity: 1000,
          },
          {
            color: "#2C4E8A",
            quantity: 624,
          },
        ],
        image:
          "https://courtcanva-image-node.s3.ap-southeast-2.amazonaws.com/preview-image/ZeLzceibuDweTAeoDk9mQ.jpeg",
        constructionDrawing: "https://developer.mozilla.org",
      },
      {
        design: {
          designName: "Court Canva 1",
          courtSize: {
            name: "Pro Full Court",
            length: 28000,
            width: 15000,
            threePointLine: 900,
            threePointRadius: 6600,
            centreCircleRadius: 1800,
            restrictedAreaLength: 5790,
            restrictedAreaWidth: 4800,
            sideBorderWidth: 1000,
            lengthOfCorner: 1575,
            lineBorderWidth: 200,
          },
          tileColor: [
            {
              location: "threePoint",
              color: "#7088B1",
            },
            {
              location: "courtArea",
              color: "#E18E11",
            },
            {
              location: "topKeyArea",
              color: "#B6B6B6",
            },
            {
              location: "border",
              color: "#834085",
            },
            {
              location: "keyArea",
              color: "#2C4E8A",
            },
            {
              location: "circleArea",
              color: "#B6B6B6",
            },
          ],
          courtType: "basketball",
        },
        quotation: "67445.40",
        quotationDetails: [
          {
            color: "#7088B1",
            quantity: 1273,
          },
          {
            color: "#E18E11",
            quantity: 2572,
          },
          {
            color: "#B6B6B6",
            quantity: 231,
          },
          {
            color: "#834085",
            quantity: 1000,
          },
          {
            color: "#2C4E8A",
            quantity: 624,
          },
        ],
        image:
          "https://courtcanva-image-node.s3.ap-southeast-2.amazonaws.com/preview-image/EssoCZLRV2Dv61Wq_ztpP.jpeg",
        constructionDrawing: "https://developer.mozilla.org",
      },
    ],
    depositRatio: 0.02,
    createdAt: "2022-11-13T11:42:53.539Z",
    updatedAt: "2022-11-13T11:59:24.466Z",
    __v: 0,
    paymentInfo: {
      _id: "6370dc1ce71fd524e1c43855",
      orderId: "6370d83de71fd524e1c437b8",
      name: "sdfas",
      email: "asdf@gmail.com",
      phone: "+61482145456",
      billingAddress: {
        city: "Footscray",
        country: "AU",
        line1: "2/30 Rosamond Road",
        line2: null,
        state: "VIC",
        postalCode: "3011",
      },
      constructionAddress: {
        city: "Footscray",
        country: "AU",
        line1: "2/30 Rosamond Road",
        line2: null,
        state: "VIC",
        postalCode: "3011",
      },
      currency: "aud",
      amountTotal: 269782,
      sessionId: "cs_test_b1JMBXdvKY0nlBornsx9fi71qonXJYMv6pjwzlatW0UBoWtng6sJtsrUVe",
      createdAt: "2022-11-13T11:59:24.459Z",
      updatedAt: "2022-11-13T11:59:24.459Z",
      __v: 0,
    },
  },
];

const mockOrdersData: IMyOrder[] = mockOrdersRowData.map((order: any) => {
  return {
    userId: order.user_id,
    _id: order._id,
    status: order.status,
    createdAt: order.createdAt,
    paidAt: order.paymentInfo ? order.paymentInfo.createdAt : "",
    consigneeName: order.paymentInfo ? order.paymentInfo.name : "",
    consigneePhoneNo: order.paymentInfo ? order.paymentInfo.phone : "",
    consigneeEmail: order.paymentInfo ? order.paymentInfo.email : "",
    shoppingAddressCity: order.paymentInfo ? order.paymentInfo.constructionAddress.city : "",
    shoppingAddressState: order.paymentInfo ? order.paymentInfo.constructionAddress.state : "",
    shoppingAddressCountry: order.paymentInfo ? order.paymentInfo.constructionAddress.country : "",
    shoppingAddressLine1: order.paymentInfo ? order.paymentInfo.constructionAddress.line1 : "",
    shoppingAddressLine2: order.paymentInfo ? order.paymentInfo.constructionAddress.line2 : "",
    shoppingAddressPostalCode: order.paymentInfo
      ? order.paymentInfo.constructionAddress.postalCode
      : "",
    currency: order.paymentInfo ? order.paymentInfo.currency : "aud",
    depositRatio: order.depositRatio,
    items: order.items,
  };
});

export default mockOrdersData;
