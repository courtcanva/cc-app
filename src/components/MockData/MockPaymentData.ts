import { PageStatus } from "@/constants/paymentResponsePage";
import { IOrderWithPaymentInfo, OrderDetails, PaymentResPage } from "@/interfaces/order";

export const mockPaymentFailurePageData: PaymentResPage = {
  status: PageStatus.FAILURE,
  orderDetails: {
    createdAt: "01/01/2022 11:11",
    paidAt: null,
    amount: "666.66",
    orderId: "123456789",
  },
};

export const mockPaymentSuccessPageData: PaymentResPage = {
  status: PageStatus.SUCCESS,
  orderDetails: {
    createdAt: "02/01/2022 11:11",
    paidAt: "03/01/2022 11:11",
    amount: "2333.33",
    orderId: "1145141919",
  },
};

export const mockPaymentErrorPageData: PaymentResPage = {
  status: PageStatus.ERROR,
};
