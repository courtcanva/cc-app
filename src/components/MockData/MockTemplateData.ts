import { ITestImyTemplates } from "@/interfaces/template";

export const mockTemplateDataRaw: any = [
  {
    _id: {
      $oid: "632a6741439020e45d11aa2a",
    },
    user_id: "117321211547215290439",
    design: {
      designer: "test designer",
      designName: "Small Court example",
      image: "",
      courtSize: {
        name: "Small Court",
        length: 9000,
        width: 5000,
        threePointLine: 900,
        threePointRadius: 6600,
        centreCircleRadius: 1800,
        restrictedAreaLength: 5790,
        restrictedAreaWidth: 4800,
        sideBorderWidth: 0,
        lengthOfCorner: 1575,
        lineBorderWidth: 200,
      },
      tileColor: [
        {
          location: "threePoint",
          color: "#7088B1",
        },
        {
          location: "courtArea",
          color: "#E18E11",
        },
        {
          location: "topKeyArea",
          color: "#B6B6B6",
        },
        {
          location: "border",
          color: "#834085",
        },
        {
          location: "keyArea",
          color: "#2C4E8A",
        },
        {
          location: "circleArea",
          color: "#B6B6B6",
        },
      ],
    },
    image:
      "https://jr-bookinglet-2.s3.ap-southeast-2.amazonaws.com/court_canva_test/m8i7liw-KKm9Dl2OYwgi_.png",
    createdAt: {
      $date: {
        $numberLong: "1663723361318",
      },
    },
    updatedAt: {
      $date: {
        $numberLong: "1663723361318",
      },
    },
    status: "Published",
    description: "loreamsdhuhihuhuihu",
    tags: {
      CourtType: "SMALLCOURT",
      CourtCategory: "BASKETBALL",
    },
    isDeleted: false,
    __v: 0,
  },
  {
    _id: {
      $oid: "632a6761439020e45d11aa2d",
    },
    user_id: "117321211547215290430",
    design: {
      designer: "test designer",
      designName: "Pro Full Court example",
      image: "",
      courtSize: {
        name: "Pro Full Court",
        length: 28000,
        width: 15000,
        threePointLine: 900,
        threePointRadius: 6600,
        centreCircleRadius: 1800,
        restrictedAreaLength: 5790,
        restrictedAreaWidth: 4900,
        sideBorderWidth: 1000,
        lengthOfCorner: 1575,
        lineBorderWidth: 200,
      },
      tileColor: [
        {
          location: "threePoint",
          color: "#7088B1",
        },
        {
          location: "courtArea",
          color: "#E18E11",
        },
        {
          location: "topKeyArea",
          color: "#B6B6B6",
        },
        {
          location: "border",
          color: "#834085",
        },
        {
          location: "keyArea",
          color: "#2C4E8A",
        },
        {
          location: "circleArea",
          color: "#B6B6B6",
        },
      ],
    },
    image:
      "https://jr-bookinglet-2.s3.ap-southeast-2.amazonaws.com/court_canva_test/m8i7liw-KKm9Dl2OYwgi_.png",
    createdAt: {
      $date: {
        $numberLong: "1663723361318",
      },
    },
    updatedAt: {
      $date: {
        $numberLong: "1663723361318",
      },
    },
    isDeleted: false,
    status: "Published",
    description: "loreamsdhuhihuhuihu",
    tags: {
      CourtType: "PROFULLCOURT",
      CourtCategory: "BASKETBALL",
    },
    __v: 0,
  },
  {
    _id: {
      $oid: "632a6761439020e45d11aa2e",
    },
    user_id: "117321211547215290431",
    design: {
      designer: "test designer",
      designName: "Pro Full Court example",
      image: "",
      courtSize: {
        name: "Pro Full Court",
        length: 28000,
        width: 15000,
        threePointLine: 900,
        threePointRadius: 6600,
        centreCircleRadius: 1800,
        restrictedAreaLength: 5790,
        restrictedAreaWidth: 4900,
        sideBorderWidth: 1000,
        lengthOfCorner: 1575,
        lineBorderWidth: 200,
      },
      tileColor: [
        {
          location: "threePoint",
          color: "#7088B1",
        },
        {
          location: "courtArea",
          color: "#E18E11",
        },
        {
          location: "topKeyArea",
          color: "#B6B6B6",
        },
        {
          location: "border",
          color: "#834085",
        },
        {
          location: "keyArea",
          color: "#2C4E8A",
        },
        {
          location: "circleArea",
          color: "#B6B6B6",
        },
      ],
    },
    image:
      "https://jr-bookinglet-2.s3.ap-southeast-2.amazonaws.com/court_canva_test/m8i7liw-KKm9Dl2OYwgi_.png",
    createdAt: {
      $date: {
        $numberLong: "1663723361318",
      },
    },
    updatedAt: {
      $date: {
        $numberLong: "1663723361318",
      },
    },
    status: "Published",
    description: "lorea msdhuhi huhu ihu",
    tags: {
      CourtType: "PROFULLCOURT",
      CourtCategory: "BASKETBALL",
    },
    isDeleted: false,
    __v: 0,
  },
];

export const mockTemplateData: ITestImyTemplates[] = mockTemplateDataRaw.map((item: any) => {
  return {
    _id: item._id,
    courtName: item.design.designName,
    user_id: item.user_id,
    description: item.description,
    image: item.image,
    status: item.status,
    createdAt: item.createdAt,
    tags: item.tags,
    design: item.design,
  };
});
