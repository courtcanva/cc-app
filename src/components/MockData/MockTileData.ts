import { AreaTileQty } from "@/store/reducer/areaTileQtySlice";

interface MockTileData {
  id: string;
  name: string;
  tileQty: AreaTileQty[];
}
export const mockTileData: MockTileData[] = [
  {
    id: "62c432cfb8a9c5f61f038320",
    name: "Full Court",
    tileQty: [
      {
        location: "courtArea",
        quantity: 2572,
      },
      {
        location: "threePoint",
        quantity: 1273,
      },
      {
        location: "keyArea",
        quantity: 624,
      },
      {
        location: "topKeyArea",
        quantity: 116,
      },
      {
        location: "circleArea",
        quantity: 115,
      },
      {
        location: "border",
        quantity: 0,
      },
    ],
  },
  {
    id: "62c432cfb8a9c5f61f03831f",
    name: "Pro Full Court",
    tileQty: [
      {
        location: "courtArea",
        quantity: 2572,
      },
      {
        location: "threePoint",
        quantity: 1273,
      },
      {
        location: "keyArea",
        quantity: 624,
      },
      {
        location: "topKeyArea",
        quantity: 116,
      },
      {
        location: "circleArea",
        quantity: 115,
      },
      {
        location: "border",
        quantity: 1000,
      },
    ],
  },
  {
    id: "62c432cfb8a9c5f61f038322",
    name: "Half Court",
    tileQty: [
      {
        location: "courtArea",
        quantity: 704,
      },
      {
        location: "threePoint",
        quantity: 632,
      },
      {
        location: "keyArea",
        quantity: 304,
      },
      {
        location: "topKeyArea",
        quantity: 60,
      },
      {
        location: "circleArea",
        quantity: 0,
      },
      {
        location: "border",
        quantity: 0,
      },
    ],
  },
  {
    id: "62c432cfb8a9c5f61f038321",
    name: "Pro Half Court",
    tileQty: [
      {
        location: "courtArea",
        quantity: 1295,
      },
      {
        location: "threePoint",
        quantity: 632,
      },
      {
        location: "keyArea",
        quantity: 304,
      },
      {
        location: "topKeyArea",
        quantity: 60,
      },
      {
        location: "circleArea",
        quantity: 59,
      },
      {
        location: "border",
        quantity: 0,
      },
    ],
  },
  {
    id: "62c432cfb8a9c5f61f038324",
    name: "Small Court",
    tileQty: [
      {
        location: "courtArea",
        quantity: 57,
      },
      {
        location: "threePoint",
        quantity: 71,
      },
      {
        location: "keyArea",
        quantity: 323,
      },
      {
        location: "topKeyArea",
        quantity: 59,
      },
      {
        location: "circleArea",
        quantity: 0,
      },
      {
        location: "border",
        quantity: 0,
      },
    ],
  },
  {
    id: "62c432cfb8a9c5f61f038323",
    name: "Medium Court",
    tileQty: [
      {
        location: "courtArea",
        quantity: 189,
      },
      {
        location: "threePoint",
        quantity: 263,
      },
      {
        location: "keyArea",
        quantity: 304,
      },
      {
        location: "topKeyArea",
        quantity: 60,
      },
      {
        location: "circleArea",
        quantity: 0,
      },
      {
        location: "border",
        quantity: 0,
      },
    ],
  },
];
