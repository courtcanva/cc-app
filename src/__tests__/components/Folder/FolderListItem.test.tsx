import { mockDesignData } from "@/components/MockData/MockDesignData";
import FolderListItem from "@/components/FolderList/FolderListItem";
import { screen } from "@testing-library/react";
import renderWithMockedProvider from "../../utils";
import userEvent from "@testing-library/user-event";

describe("Render folder list", () => {
  it("Should render folder list item correctly", () => {
    const funMock = jest.fn();
    const { container } = renderWithMockedProvider(
      <FolderListItem design={mockDesignData[0]} handleCourtSelecting={funMock} />
    );
    expect(container).toMatchSnapshot();
    const item = screen.getByTestId("folderListItems-0");
    userEvent.click(item);
    expect(funMock).toBeCalled();
  });

  it("Should render opacity correctly", () => {
    const { container } = renderWithMockedProvider(
      <FolderListItem design={mockDesignData[0]} activateDesign="folderListItems - 1" />
    );
    expect(container).toMatchSnapshot();
    // todo: add 0.4 version
  });
});
