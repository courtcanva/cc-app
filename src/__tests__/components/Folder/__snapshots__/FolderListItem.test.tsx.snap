// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Render folder list Should render folder list item correctly 1`] = `
<div>
  <div
    class="css-1gsyri3"
    data-testid="folderListItems-0"
  >
    <div
      class="chakra-stack css-16awuyr"
    >
      <div
        class="css-0"
        style="background: none; border-radius: 0; min-width: 140px; margin-top: 5px;"
      >
        <div
          class="css-0"
          style="font-size: 13px; background: none; border-radius: 0; font-weight: bold;"
        >
          Court Canva 0
        </div>
        <div
          class="css-0"
          style="font-size: 9px; color: rgb(44, 82, 130); background: none; border-radius: 0;"
        >
          created at 
          02/11/2022
        </div>
      </div>
      <button
        aria-label="delete button"
        class="chakra-button css-198bfjk"
        type="button"
      >
        <button
          aria-label="Delete Button"
          class="chakra-button css-198bfjk"
          data-testid="delete-btn"
          type="button"
        >
          <svg
            aria-hidden="true"
            fill="currentColor"
            focusable="false"
            height="1em"
            stroke="currentColor"
            stroke-width="0"
            viewBox="0 0 448 512"
            width="1em"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M32 464a48 48 0 0 0 48 48h288a48 48 0 0 0 48-48V128H32zm272-256a16 16 0 0 1 32 0v224a16 16 0 0 1-32 0zm-96 0a16 16 0 0 1 32 0v224a16 16 0 0 1-32 0zm-96 0a16 16 0 0 1 32 0v224a16 16 0 0 1-32 0zM432 32H312l-9.4-18.7A24 24 0 0 0 281.1 0H166.8a23.72 23.72 0 0 0-21.4 13.3L136 32H16A16 16 0 0 0 0 48v32a16 16 0 0 0 16 16h416a16 16 0 0 0 16-16V48a16 16 0 0 0-16-16z"
            />
          </svg>
        </button>
      </button>
    </div>
    <div
      class="css-r0om48"
    >
      <span
        style="box-sizing: border-box; display: block; overflow: hidden; background: none; opacity: 1; border: 0px; margin: 0px; padding: 0px; position: absolute; top: 0px; left: 0px; bottom: 0px; right: 0px;"
      >
        <img
          alt="Court image"
          data-nimg="fill"
          decoding="async"
          src="data:image/gif;base64,R0lGODlhAQABAIAAAAAAAP///yH5BAEAAAAALAAAAAABAAEAAAIBRAA7"
          style="position: absolute; top: 0px; left: 0px; bottom: 0px; right: 0px; box-sizing: border-box; padding: 0px; margin: auto; display: block; width: 0px; height: 0px; min-width: 100%; max-width: 100%; min-height: 100%; max-height: 100%; object-fit: contain;"
        />
        <noscript />
      </span>
    </div>
    <div
      class="chakra-stack css-16awuyr"
    >
      <span
        class="css-10kxp63"
        style="border-radius: 0; font-size: 8px; margin-bottom: 5px; font-weight: bold; text-align: center; padding: 0px 16px;"
      >
        full court
      </span>
      <span
        class="css-3uf65l"
        style="border-radius: 0; font-size: 8px; margin-bottom: 5px; padding: 0px 16px; font-weight: bold;"
      >
        Basketball
      </span>
    </div>
  </div>
</div>
`;

exports[`Render folder list Should render opacity correctly 1`] = `
<div>
  <div
    class="css-kvs84j"
    data-testid="folderListItems-0"
  >
    <div
      class="chakra-stack css-16awuyr"
    >
      <div
        class="css-0"
        style="background: none; border-radius: 0; min-width: 140px; margin-top: 5px;"
      >
        <div
          class="css-0"
          style="font-size: 13px; background: none; border-radius: 0; font-weight: bold;"
        >
          Court Canva 0
        </div>
        <div
          class="css-0"
          style="font-size: 9px; color: rgb(44, 82, 130); background: none; border-radius: 0;"
        >
          created at 
          02/11/2022
        </div>
      </div>
      <button
        aria-label="delete button"
        class="chakra-button css-198bfjk"
        type="button"
      >
        <button
          aria-label="Delete Button"
          class="chakra-button css-198bfjk"
          data-testid="delete-btn"
          type="button"
        >
          <svg
            aria-hidden="true"
            fill="currentColor"
            focusable="false"
            height="1em"
            stroke="currentColor"
            stroke-width="0"
            viewBox="0 0 448 512"
            width="1em"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M32 464a48 48 0 0 0 48 48h288a48 48 0 0 0 48-48V128H32zm272-256a16 16 0 0 1 32 0v224a16 16 0 0 1-32 0zm-96 0a16 16 0 0 1 32 0v224a16 16 0 0 1-32 0zm-96 0a16 16 0 0 1 32 0v224a16 16 0 0 1-32 0zM432 32H312l-9.4-18.7A24 24 0 0 0 281.1 0H166.8a23.72 23.72 0 0 0-21.4 13.3L136 32H16A16 16 0 0 0 0 48v32a16 16 0 0 0 16 16h416a16 16 0 0 0 16-16V48a16 16 0 0 0-16-16z"
            />
          </svg>
        </button>
      </button>
    </div>
    <div
      class="css-r0om48"
    >
      <span
        style="box-sizing: border-box; display: block; overflow: hidden; background: none; opacity: 1; border: 0px; margin: 0px; padding: 0px; position: absolute; top: 0px; left: 0px; bottom: 0px; right: 0px;"
      >
        <img
          alt="Court image"
          data-nimg="fill"
          decoding="async"
          src="data:image/gif;base64,R0lGODlhAQABAIAAAAAAAP///yH5BAEAAAAALAAAAAABAAEAAAIBRAA7"
          style="position: absolute; top: 0px; left: 0px; bottom: 0px; right: 0px; box-sizing: border-box; padding: 0px; margin: auto; display: block; width: 0px; height: 0px; min-width: 100%; max-width: 100%; min-height: 100%; max-height: 100%; object-fit: contain;"
        />
        <noscript />
      </span>
    </div>
    <div
      class="chakra-stack css-16awuyr"
    >
      <span
        class="css-10kxp63"
        style="border-radius: 0; font-size: 8px; margin-bottom: 5px; font-weight: bold; text-align: center; padding: 0px 16px;"
      >
        full court
      </span>
      <span
        class="css-3uf65l"
        style="border-radius: 0; font-size: 8px; margin-bottom: 5px; padding: 0px 16px; font-weight: bold;"
      >
        Basketball
      </span>
    </div>
  </div>
</div>
`;
